external/storm-hbase/src/main/java/org/apache/storm/hbase/bolt/HBaseBolt.java

```diff
@@ -53,21 +61,62 @@ public HBaseBolt withConfigKey(String configKey) {
         return this;
     }
 
+    public HBaseBolt withBatchSize(int batchSize) {
+        this.batchSize = batchSize;
+        return this;
+    }
+
+    public HBaseBolt withFlushIntervalSecs(int flushIntervalSecs) {
+        this.flushIntervalSecs = flushIntervalSecs;
+        return this;
+    }
+
+    @Override
+    public Map<String, Object> getComponentConfiguration() {
+        Map<String, Object> conf = super.getComponentConfiguration();
+        if (conf == null)
+            conf = new Config();
+
+        if (flushIntervalSecs > 0) {
+            LOG.info("Enabling tick tuple with interval [" + flushIntervalSecs + "]");
+            conf.put(Config.TOPOLOGY_TICK_TUPLE_FREQ_SECS, flushIntervalSecs);
+        }
+
+        return conf;
+    }
+
+
     @Override
     public void execute(Tuple tuple) {
-        byte[] rowKey = this.mapper.rowKey(tuple);
-        ColumnList cols = this.mapper.columns(tuple);
-        List<Mutation> mutations = hBaseClient.constructMutationReq(rowKey, cols, writeToWAL? Durability.SYNC_WAL : Durability.SKIP_WAL);
+        boolean flush = false;
+        if (TupleUtils.isTick(tuple)) {
+            LOG.debug("TICK received! current batch status [" + tupleBatch.size() + "/" + batchSize + "]");
+            flush = true;
+        } else {
+            byte[] rowKey = this.mapper.rowKey(tuple);
+            ColumnList cols = this.mapper.columns(tuple);
+            List<Mutation> mutations = hBaseClient.constructMutationReq(rowKey, cols, writeToWAL? Durability.SYNC_WAL : Durability.SKIP_WAL);
+            batchMutations.addAll(mutations);
+            tupleBatch.add(tuple);
+            if (tupleBatch.size() >= batchSize)
```

Comment:
Minor nit, but if/for blocks should include curly braces even if they're only one line.